presto-orc/src/test/java/com/facebook/presto/orc/IntegerDictionaryProviderTest.java

```diff
@@ -0,0 +1,302 @@
+import com.facebook.presto.orc.metadata.OrcType;
+import com.facebook.presto.orc.reader.IntegerDictionaryProvider;
+import com.facebook.presto.orc.stream.InputStreamSource;
+import com.facebook.presto.orc.stream.InputStreamSources;
+import com.facebook.presto.orc.stream.LongInputStreamDwrf;
+import com.facebook.presto.orc.stream.LongOutputStreamDwrf;
+import com.facebook.presto.orc.stream.OrcInputStream;
+import com.facebook.presto.orc.stream.SharedBuffer;
+import com.facebook.presto.orc.stream.StreamDataOutput;
+import com.facebook.presto.orc.stream.ValueInputStream;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.slice.Slice;
+import io.airlift.units.DataSize;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.orc.OrcDecompressor.createOrcDecompressor;
+import static com.facebook.presto.orc.checkpoint.Checkpoints.getDictionaryStreamCheckpoint;
+import static com.facebook.presto.orc.metadata.CompressionKind.SNAPPY;
+import static com.facebook.presto.orc.metadata.OrcType.OrcTypeKind.LONG;
+import static com.facebook.presto.orc.metadata.Stream.StreamKind.DICTIONARY_DATA;
+import static com.facebook.presto.orc.stream.CheckpointInputStreamSource.createCheckpointStreamSource;
+import static io.airlift.units.DataSize.Unit.KILOBYTE;
+import static java.lang.Math.toIntExact;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotEquals;
+import static org.testng.Assert.assertTrue;
+
+public class IntegerDictionaryProviderTest
+{
+    private static final OrcDataSourceId ORC_DATA_SOURCE_ID = new OrcDataSourceId("dict_provider_test");
+    private static final DataSize COMPRESSION_BLOCK_SIZE = new DataSize(256, KILOBYTE);
+    private static final OrcType LONG_TYPE = new OrcType(LONG, ImmutableList.of(), ImmutableList.of(), Optional.empty(), Optional.empty(), Optional.empty());
+    private static final DummyOrcDataSource DUMMY_ORC_DATA_SOURCE = new DummyOrcDataSource();
+
+    @Test(dataProvider = "dataForDictionaryLoadingTest")
+    public void testIntegerDictionaryLoading(Map<NodeId, long[]> dictionaryStreams, List<NodeId> unused)
+            throws Exception
+    {
+        IntegerDictionaryProvider dictionaryProvider = new IntegerDictionaryProvider(createLongDictionaryStreamSources(dictionaryStreams));
+        for (Map.Entry<NodeId, long[]> entry : dictionaryStreams.entrySet()) {
+            StreamId streamId = entry.getKey().toDictionaryDataStreamId();
+            long[] data = entry.getValue();
+            long[] dictionary = new long[0];
+            long[] newDictionary = dictionaryProvider.getDictionary(createFlatStreamDescriptor(streamId), dictionary, data.length);
+            assertNotEquals(newDictionary, dictionary);
```

Comment:
I think this check can be removed, because `assertEquals(newDictionary, data)` already does all the work.

Reply:
I meant to check the objects per se are different, but now this is checked in buffer reuse test instead.